### PR TITLE
Document C++ Initializers

### PR DESCRIPTION
**Pull request by jmgomez**

### CODE REVIEW

File: doc/manual_experimental.md [hunk 1]
```diff
@@ -2382,6 +2382,31 @@ In the example above `CppClass` has a deleted default constructor. Notice how by
...
+  int x;
+  char* y;
+};
+""".}
+type
+  CppStruct {.importcpp, inheritable.} = object
+
+proc constructCppStruct(a:cint = 5, b:cstring = "hello"): CppStruct {.importcpp: "CppStruct(@)", constructor.}
+
+
```
Comment (juancarlospaco) [inline doc/manual_experimental.md:2409]: Add an actual use kinda like an `assert` or `echo` or something like that,
users sometimes complain they copypaste examples and wont run or "does nothing" etc IMHO.

File: doc/manual_experimental.md [hunk 2]
```diff
@@ -2382,6 +2382,33 @@ In the example above `CppClass` has a deleted default constructor. Notice how by
...
+```nim
+
+{.emit:"""/*TYPESECTION*/
+struct CppStruct {
+  CppStruct(int x, char* y): x(x), y(y){}
+  int x;
+  char* y;
+};
+""".}
```
Comment (Varriount) [inline doc/manual_experimental.md]:
```suggestion
{.emit:"""
/*TYPESECTION*/
struct CppStruct {
  CppStruct(int x, char* y): x(x), y(y){}
  int x;
  char* y;
};
""".}
```

File: doc/manual_experimental.md [hunk 3]
```diff
@@ -2382,6 +2382,33 @@ In the example above `CppClass` has a deleted default constructor. Notice how by
...
+type
+  CppStruct {.importcpp, inheritable.} = object
+
+proc makeCppStruct(a:cint = 5, b:cstring = "hello"): CppStruct {.importcpp: "CppStruct(@)", constructor.}
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
proc makeCppStruct(a: cint = 5, b:cstring = "hello"): CppStruct {.importcpp: "CppStruct(@)", constructor.}
```

File: doc/manual_experimental.md [hunk 4]
```diff
@@ -2382,6 +2382,33 @@ In the example above `CppClass` has a deleted default constructor. Notice how by
...
+
+(proc (s:CppStruct) = echo "hello")(makeCppStruct())
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
(proc (s: CppStruct) = echo "hello")(makeCppStruct()) 
```

File: doc/manual_experimental.md [hunk 5]
```diff
@@ -2382,6 +2382,33 @@ In the example above `CppClass` has a deleted default constructor. Notice how by
...
+  char* y;
+};
+""".}
+type
+  CppStruct {.importcpp, inheritable.} = object
+
+proc makeCppStruct(a:cint = 5, b:cstring = "hello"): CppStruct {.importcpp: "CppStruct(@)", constructor.}
+
+(proc (s:CppStruct) = echo "hello")(makeCppStruct()) 
+#if one remove a default value from the constructor and pass it to the call. The C++ compiler will complain.
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
# If one removes a default value from the constructor and passes it to the call explicitly, the C++ compiler will complain.
```

File: doc/manual_experimental.md [hunk 6]
```diff
@@ -2382,6 +2382,33 @@ In the example above `CppClass` has a deleted default constructor. Notice how by
...
+=======================
+
+By default Nim initializes `importcpp` types with `{}`. This can be problematic when importing
+types with a deleted default constructor. In order to avoid this, one can specify default values for a constructor by specifying default values for the proc params in the `constructor` proc.
+
+For example:
+
+```nim
+
+{.emit:"""/*TYPESECTION*/
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
{.emit: """/*TYPESECTION*/
```

**Pull request merged**